bolt-modules/boltlib/lib/puppet/functions/apply_prep.rb

```diff
@@ -60,18 +62,25 @@ def executor
     @executor ||= Puppet.lookup(:bolt_executor)
   end
 
-  def apply_prep(target_spec)
+  def apply_prep(target_spec, options = {})
     unless Puppet[:tasks]
       raise Puppet::ParseErrorWithIssue
         .from_issue_and_stack(Bolt::PAL::Issues::PLAN_OPERATION_NOT_SUPPORTED_WHEN_COMPILING, action: 'apply_prep')
     end
 
+    options = options.transform_keys { |k| k.sub(/^_/, '').to_sym }
+
     applicator = Puppet.lookup(:apply_executor)
 
     executor.report_function_call(self.class.name)
 
     targets = inventory.get_targets(target_spec)
 
+    @required_modules = Array(options[:required_modules])
+    if @required_modules.any?
```

Comment:
```suggestion
    if @required_modules&.any?
```
With the changes above this may be nil, in which case the if will still correctly evaluate to false.